runtime/gc_vlhgc/VLHGCAccessBarrier.cpp

```diff
@@ -271,6 +271,20 @@ MM_VLHGCAccessBarrier::copyArrayCritical(J9VMThread *vmThread, GC_ArrayObjectMod
 	}
 }
 
+IDATA
+MM_VLHGCAccessBarrier::indexableDataDisplacement(J9VMThread *vmThread,  J9IndexableObject *src,  J9IndexableObject *dst)
+{
+	IDATA displacement = 0;
+	GC_ArrayObjectModel *indexableObjectModel = &_extensions->indexableObjectModel;
+
+	Assert_MM_true((0 == indexableObjectModel->getSizeInElements(dst)) || indexableObjectModel->isInlineContiguousArraylet(dst));
+
```

Comment:
add a comment before adjacency check , that we are doing it against dst object since src object may be overwritten during sliding compaction